src/main/java/io/cryostat/platform/internal/OpenShiftPlatformStrategy.java

```diff
@@ -72,32 +65,12 @@ public int getPriority() {
     }
 
     @Override
-    public boolean isAvailable() {
-        logger.trace("Testing OpenShift Platform Availability");
-        if (osClient == null) {
-            return false;
-        }
-        try {
-            String namespace = getOwnNamespace();
-            if (namespace == null) {
-                return false;
-            }
-            osClient.routes().inNamespace(namespace).list();
-            return true;
-        } catch (Exception e) {
-            logger.info(e);
-            return false;
-        }
-    }
-
-    @Override
-    public KubeApiPlatformClient getPlatformClient() {
-        logger.info("Selected OpenShift Platform Strategy");
-        return super.getPlatformClient();
+    protected BiFunction<OpenShiftClient, String, ?> testAvailability() {
+        return (client, ns) -> client.routes().inNamespace(ns).list();
```

Comment:
I wonder if we'd still run into problems if Cryostat can't list routes in its own namespace. Maybe this method would work instead? https://github.com/fabric8io/kubernetes-client/blob/188bd37280ca96ec2a84d5a7bb088e4387f687b3/openshift-client/src/main/java/io/fabric8/openshift/client/impl/OpenShiftClientImpl.java#L736-L742